webpack/ForemanTasks/Components/TaskDetails/TaskDetailsActions.js

```diff
@@ -1,143 +1,75 @@
-import { API } from 'foremanReact/redux/API';
-import {
-  showLoading,
-  hideLoading,
-} from 'foremanReact/components/Layout/LayoutActions';
+import { post, get } from 'foremanReact/redux/API';
 import { addToast } from 'foremanReact/redux/actions/toasts';
 import { translate as __ } from 'foremanReact/common/I18n';
 import {
-  FOREMAN_TASK_DETAILS_FETCH_TASK_REQUEST,
-  FOREMAN_TASK_DETAILS_FETCH_TASK_SUCCESS,
-  FOREMAN_TASK_DETAILS_FETCH_TASK_FAILURE,
-  FOREMAN_TASK_DETAILS_STOP_POLLING,
-  FOREMAN_TASK_DETAILS_START_POLLING,
-  TASK_STEP_CANCEL_REQUEST,
-  TASK_STEP_CANCEL_FAILURE,
-  TASK_STEP_CANCEL_SUCCESS,
-} from './TaskDetailsConstants';
+  withInterval,
+  stopInterval,
+} from 'foremanReact/redux/middlewares/IntervalMiddleware';
+import { TASK_STEP_CANCEL, FOREMAN_TASK_DETAILS } from './TaskDetailsConstants';
 import {
   errorToastData,
   infoToastData,
   successToastData,
 } from '../common/ToastsHelpers';
 
-export const taskReloadStop = timeoutId => {
-  if (timeoutId) {
-    clearTimeout(timeoutId);
-  }
-  return {
-    type: FOREMAN_TASK_DETAILS_STOP_POLLING,
-    payload: { timeoutId },
-  };
-};
+export const taskReloadStop = () => stopInterval(FOREMAN_TASK_DETAILS);
 
-export const taskReloadStart = (
-  timeoutId,
-  refetchTaskDetails,
-  id,
-  loading = false
-) => {
-  if (!timeoutId) {
-    timeoutId = setInterval(
-      () => refetchTaskDetails(id, loading, timeoutId),
+export const taskReloadStart = id => dispatch => {
+  dispatch(
+    withInterval(
+      get({
+        key: FOREMAN_TASK_DETAILS,
+        url: `/foreman_tasks/api/tasks/${id}/details?include_permissions`,
+        handleSuccess: ({ data }) => {
+          if (data.state === 'stopped') {
+            dispatch(taskReloadStop(id));
+          }
+        },
+        handleError: () => {
+          dispatch(taskReloadStop());
+        },
```

Comment:
You can use the second callback param which stops the interval
```suggestion
        handleError: (error, stopTaskInterval) => stopTaskInterval(),
```